spark/src/main/java/org/apache/iceberg/actions/RemoveOrphanFilesAction.java

```diff
@@ -161,15 +172,44 @@ public RemoveOrphanFilesAction deleteWith(Consumer<String> newDeleteFunc) {
         .as(Encoders.STRING())
         .collectAsList();
 
-    Tasks.foreach(orphanFiles)
-        .noRetry()
-        .suppressFailureWhenFinished()
-        .onFailure((file, exc) -> LOG.warn("Failed to delete file: {}", file, exc))
-        .run(deleteFunc::accept);
+    paralleExecutor(sparkContext, orphanFiles, excutorTaskNum, deleteFunc);
 
     return orphanFiles;
   }
 
+  private void paralleExecutor(JavaSparkContext javaSc, List<String> orphanFiles,
+                              int numSlices, Consumer deleteConsumer) {
+    if (numSlices == 0) {
+      Tasks.foreach(orphanFiles)
+          .noRetry()
+          .suppressFailureWhenFinished()
```

Comment:
We could just make this parallel as well as in the RemoveSnapshots code, where the user can pass in a different execution pool.